meadow-client/src/Marlowe/Parser.purs

```diff
@@ -0,0 +1,149 @@
+module Marlowe.Parser where
+
+import Prelude
+
+import Control.Alternative ((<|>))
+import Data.BigInteger (BigInteger)
+import Data.BigInteger as BigInteger
+import Data.List (List)
+import Data.Maybe (Maybe(..))
+import Data.Newtype (wrap)
+import Semantics (BlockNumber, Choice, Contract(..), IdAction, IdChoice, IdCommit, IdOracle, LetLabel, Observation(..), Person, Timeout, Value(..))
+import Text.Parsing.Simple (Parser, char, fail, fix, integral, parens, some, space, string)
+
+-- All arguments are space separated so we add **> to reduce boilerplate
+
+spaces :: Parser String (List Char)
+spaces = some space
+
+appRSpaces :: forall a b. Parser String a -> Parser String b -> Parser String b
+appRSpaces p q = p *> spaces *> q
+
+infixl 4 appRSpaces as **>
+
+appSpaces :: forall a b. Parser String (a -> b) -> Parser String a -> Parser String b
+appSpaces p q = p <*> (spaces *> q)
+
+infixl 4 appSpaces as <**>
+
+------------------------------------------------------------------------
+
+bigInteger :: Parser String BigInteger
+bigInteger = do
+    i <- integral
+    case BigInteger.fromString i of
+      (Just v) -> pure v
+      Nothing -> fail "not a valid BigInt"
+
+mkIdChoice :: BigInteger -> BigInteger -> IdChoice
+mkIdChoice choice person = wrap { choice, person }
+
+idChoice :: Parser String IdChoice
+idChoice = parens (mkIdChoice <$> bigInteger <* char ',' <* space <*> bigInteger)
+
+choice :: Parser String Choice
+choice = bigInteger
+
+blockNumber :: Parser String BlockNumber
+blockNumber = bigInteger
+
+timeout :: Parser String Timeout
+timeout = blockNumber
+
+idOracle :: Parser String IdOracle
+idOracle = bigInteger
+
+idCommit :: Parser String IdCommit
+idCommit = bigInteger
```

Comment:
These make me wish that we used `newtype`s rather than aliases for all these `BigInteger`s. But perhaps that's a future improvement rather than one for this PR.

Reply:
future improvement in another PR, it is planned (in my head and I mentioned to @palas )